File: server/src/com/broll/gainea/server/core/goals/impl/e2/G_AllMountains.java
package com.broll.gainea.server.core.goals.impl.e2;

import com.broll.gainea.server.core.goals.OccupyGoal;
import com.broll.gainea.server.core.goals.GoalDifficulty;
import com.broll.gainea.server.core.map.AreaType;
import com.broll.gainea.server.core.map.ExpansionType;

public class G_AllMountains extends OccupyGoal {
    public G_AllMountains() {
        super(GoalDifficulty.EASY, "Erobere alle Berge");
    }

    @Override
    protected void initOccupations() {
        occupy(it -> it.getType() == AreaType.MOUNTAIN, ExpansionType.ICELANDS);
    }

}

File: server/src/main/com/broll/gainea/server/core/goals/impl/e2/G_AllMountains.kt
package com.broll.gainea.server.core.goals.impl.e2

import com.broll.gainea.server.core.goals.GoalDifficultyimport

com.broll.gainea.server.core.goals.OccupyGoalimport com.broll.gainea.server.core.map.Areaimport com.broll.gainea.server.core.map.AreaTypeimport com.broll.gainea.server.core.map.ExpansionType
class G_AllMountains : OccupyGoal(GoalDifficulty.EASY, "Erobere alle Berge") {
    override fun initOccupations() {
        occupy({ it: Area? -> it.getType() == AreaType.MOUNTAIN }, ExpansionType.ICELANDS)
    }
}
